test/fakeintake/server/server.go

```diff
@@ -56,13 +60,33 @@ func NewServer(options ...func(*Server)) *Server {
 		store:     serverstore.NewStore(),
 	}
 
+	registry := prometheus.NewRegistry()
+
+	registry.MustRegister(
+		collectors.NewBuildInfoCollector(),
+		collectors.NewGoCollector(),
+		collectors.NewProcessCollector(collectors.ProcessCollectorOpts{}),
+		fi.store.NbPayloads,
+	)
+
 	mux := http.NewServeMux()
 	mux.HandleFunc("/", fi.handleDatadogRequest)
 	mux.HandleFunc("/fakeintake/payloads", fi.handleGetPayloads)
 	mux.HandleFunc("/fakeintake/health", fi.handleFakeHealth)
 	mux.HandleFunc("/fakeintake/routestats", fi.handleGetRouteStats)
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution. -->
**❓ question**
Should we remove this or deprecate it, now that we have the prometheus endpoint ?

Reply:
If we are confident that nothing is using this endpoint, I agree it’s better to not keep redundant code to maintain.
Here is the PR to do the cleanup: #21214.